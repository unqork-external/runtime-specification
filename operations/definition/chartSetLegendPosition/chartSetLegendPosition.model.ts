import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { ChartSetLegendPositionOperationOptions } from './chartSetLegendPosition.options'
import { OpsBuilderStable, Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@Stable()
@OpsBuilderStable()
@Description('Updates the legend position on a Chart component.')
@DiscriminatorValue(OperationTypes.CHART_SET_LEGEND_POSITION)
export class ChartSetLegendPositionOperation extends Operation<OperationTypes.CHART_SET_LEGEND_POSITION> {
  @Required()
  public type = OperationTypes.CHART_SET_LEGEND_POSITION as const

  @Required()
  public options: ChartSetLegendPositionOperationOptions = new ChartSetLegendPositionOperationOptions()
}
