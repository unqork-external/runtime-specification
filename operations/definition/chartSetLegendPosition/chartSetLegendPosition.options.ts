import { Description, Enum, Optional, Required } from '@tsed/schema'

import { ChartAlignments, ChartVerticalAlignments } from '../../../components'

export class ChartSetLegendPositionOperationOptions {
  @Required()
  @Description('The Property ID for the target Chart component.')
  targetKey: string

  @Optional()
  @Enum(ChartAlignments)
  @Description("The horizontal alignment of the Legend relative to the Chart component's area.")
  alignment?: ChartAlignments

  @Optional()
  @Enum(ChartVerticalAlignments)
  @Description("The vertical alignment of the Legend relative to the Chart component's area.")
  verticalAlignment?: ChartVerticalAlignments
}
