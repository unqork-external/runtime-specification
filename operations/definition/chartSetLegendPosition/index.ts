export * from './chartSetLegendPosition.model'
export * from './chartSetLegendPosition.options'
