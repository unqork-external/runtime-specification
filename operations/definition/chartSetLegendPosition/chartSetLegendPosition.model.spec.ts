import { ChartSetLegendPositionOperation } from './chartSetLegendPosition.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ChartSetLegendPosition Tests', () => {
  it('ChartSetLegendPosition Model', () => {
    const { schema } = generateSchemaAndValidate(ChartSetLegendPositionOperation)

    expect(schema).toMatchSnapshot()
  })
})
