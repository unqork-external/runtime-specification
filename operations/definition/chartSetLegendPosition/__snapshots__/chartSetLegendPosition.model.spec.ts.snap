// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: ChartSetLegendPosition Tests ChartSetLegendPosition Model 1`] = `
{
  "definitions": {
    "ChartSetLegendPositionOperationOptions": {
      "properties": {
        "alignment": {
          "description": "The horizontal alignment of the Legend relative to the Chart component's area.",
          "enum": [
            "left",
            "center",
            "right",
          ],
          "type": "string",
        },
        "targetKey": {
          "description": "The Property ID for the target Chart component.",
          "minLength": 1,
          "type": "string",
        },
        "verticalAlignment": {
          "description": "The vertical alignment of the Legend relative to the Chart component's area.",
          "enum": [
            "top",
            "middle",
            "bottom",
          ],
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "description": "Updates the legend position on a Chart component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ChartSetLegendPositionOperationOptions",
    },
    "type": {
      "const": "CHART_SET_LEGEND_POSITION",
      "examples": [
        "CHART_SET_LEGEND_POSITION",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
